pkg/operator/controllers/machine/machine_controller_test.go

```diff
@@ -278,3 +188,95 @@ func TestMachineReconciler(t *testing.T) {
 		})
 	}
 }
+
+func masterMachine(name, diskSize, imagePublisher, vmSize, offer string) *machinev1beta1.Machine {
+	if diskSize == "" {
+		diskSize = "512"
+	}
+	if imagePublisher == "" {
+		imagePublisher = "azureopenshift"
+	}
+	if vmSize == "" {
+		vmSize = "Standard_D8s_v3"
+	}
+	if offer == "" {
+		offer = "aro4"
+	}
+
+	return &machinev1beta1.Machine{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: machineSetsNamespace,
+			Labels:    map[string]string{"machine.openshift.io/cluster-api-machine-role": "master"},
+		},
+		Spec: machinev1beta1.MachineSpec{
+			ProviderSpec: machinev1beta1.ProviderSpec{
+				Value: &runtime.RawExtension{
+					Raw: []byte(fmt.Sprintf(`{
+"apiVersion": "azureproviderconfig.openshift.io/v1beta1",
+"kind": "AzureMachineProviderSpec",
+"osDisk": {
+"diskSizeGB": %v
+},
+"image": {
+"publisher": "%v",
+"offer": "%v"
+},
+"vmSize": "%v"
+}`, diskSize, imagePublisher, offer, vmSize))},
+			},
+		},
+	}
+}
+
+func workerMachine(name, diskSize, imagePublisher, vmSize, offer string) *machinev1beta1.Machine {
+	if diskSize == "" {
+		diskSize = "128"
+	}
+	if imagePublisher == "" {
+		imagePublisher = "azureopenshift"
+	}
+	if vmSize == "" {
+		vmSize = "Standard_D4s_v3"
+	}
+	if offer == "" {
+		offer = "aro4"
+	}
+
+	return &machinev1beta1.Machine{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: machineSetsNamespace,
+			Labels:    map[string]string{"machine.openshift.io/cluster-api-machine-role": "worker"},
+		},
+		Spec: machinev1beta1.MachineSpec{
+			ProviderSpec: machinev1beta1.ProviderSpec{
+				Value: &runtime.RawExtension{
+					Raw: []byte(fmt.Sprintf(`{
+"apiVersion": "azureproviderconfig.openshift.io/v1beta1",
+"kind": "AzureMachineProviderSpec",
+"osDisk": {
+"diskSizeGB": %v
+},
+"image": {
+"publisher": "%v",
+"offer": "%v"
+},
+"vmSize": "%v"
+}`, diskSize, imagePublisher, offer, vmSize))},
+			},
+		},
+	}
+}
+
+func workerMachineSet(name string, replicas int32) *machinev1beta1.MachineSet {
```

Comment:
The value passed for `replicas` in these test cases is always `1`, so I would remove it as a parameter and just hardcode it.